File: examples/javaone-2026/src/main/java/org/example/koog/java/endpoints/ApiController.java
package org.example.koog.java.endpoints;

import org.example.koog.java.agents.KoogAgentService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.security.Principal;

@RestController
@RequestMapping("/api")
class ApiController {

    private final KoogAgentService agentService;

    public ApiController(KoogAgentService agentService) {
        this.agentService = agentService;
    }

    @PostMapping("/support")
    public ResponseEntity<AgentResponse> launchSupportAgent(
            Principal principal,
            @RequestBody SupportRequest request
    ) {
        var agentId = agentService.launchSupportAgent(principal.getName(), request.question());
        return ResponseEntity.ok(new AgentResponse(agentId));
    }


    public record SupportRequest(String question) {
    }

    public record AgentResponse(String agentId) {
    }
}

File: examples/spring-io-2026/kotlin/src/main/kotlin/org/example/koog/spring/kotlin/endpoints/ApiController.kt
package org.example.koog.spring.kotlin.endpoints

import org.example.koog.spring.kotlin.agents.customersupport.BankingAgentService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController
import java.security.Principal

@RestController
class ChatController(
    val agentService: BankingAgentService,
) {

    @PostMapping(value = ["/support"])
    suspend fun customerSupport(
        principal: Principal,
        @RequestBody request: ChatRequest
    ): ChatResponse {
        val response = agentService.launchSupportAgent(principal.getName(), request.question)
        return ChatResponse(response)
    }
}

data class ChatRequest(val question: String)

data class ChatResponse(val response: String)
